airbyte-cdk/python/airbyte_cdk/sources/file_based/config/abstract_file_based_spec.py

```diff
@@ -0,0 +1,69 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import json
+import re
+from abc import abstractmethod
+from typing import Any, Dict, List
+
+from airbyte_cdk.sources.file_based.config.file_based_stream_config import FileBasedStreamConfig
+from jsonschema import RefResolver
+from pydantic import AnyUrl, BaseModel, Field
+
+
+class AbstractFileBasedSpec(BaseModel):
+    """
+    Used during spec; allows the developer to configure the cloud provider specific options
+    that are needed when users configure a file-based source.
+    """
+
+    streams: List[FileBasedStreamConfig] = Field(
+        title="The list of streams to sync",
+        description="Streams defines the behavior for grouping files together that will be synced to the downstream destination. Each "
+        "stream has it own independent configuration to handle which files to sync, how files should be parsed, and the "
+        "validation of records against the schema.",
+        order=10,
```

Comment:
is order=10 to ensure this is the last field?

Reply:
yeah, it could go higher, but i just padded it with 10 similar to the existing s3 source where the abstract fields common to all sources start after the source specific ones